pkg/registry/apis/datasource/standalone.go

```diff
@@ -19,33 +97,190 @@ func NewStandaloneDatasource(group string) (*DataSourceAPIBuilder, error) {
 	if group != "testdata.datasource.grafana.app" {
 		return nil, fmt.Errorf("only testadata is currently supported")
 	}
-
-	orgId := int64(1)
 	pluginId := "grafana-testdata-datasource"
-	now := time.Now()
-	dss := []*datasources.DataSource{
-		{
-			OrgID:   orgId, // default -- used in the list command
-			Type:    pluginId,
-			UID:     "builtin", // fake for now
-			Created: now,
-			Updated: now,
-			Name:    "Testdata (builtin)",
-		},
-		{
-			OrgID:   orgId, // default -- used in the list command
-			Type:    pluginId,
-			UID:     "PD8C576611E62080A", // match the gdev version
-			Created: now,
-			Updated: now,
-			Name:    "gdev-testdata",
-		},
+
+	cfg, err := setting.NewCfgFromArgs(setting.CommandLineArgs{
```

Comment:
rather than having all this inside standalone... what about adding a new/simple interface that only implements the functions we use/need.

Then we can hide the fact that is delegating to the whole wire chain, and in cloud replace it with something that reads elsewhere.

Reply:
yep, that's the other option. Not sure how much we can remove if we do that, worth checking

Reply:
By new interface, i mean something like this (still VERY MUCH a WIP!):
https://github.com/grafana/grafana/pull/80325/files#diff-afe22ef7c90a7749b8d0d7f78a47d27a27b9be0b748bf5c0db47520d44cecd35R15

Behind that interface, i can implement two approaches:
1. wired up to all our dependencies as they are now
2. using the k8s client-go to delegate queries to the *.datasource.grafana.app servers

Reply:
That's a somewhat crazy dependency graph :) Separate interface is def on our agenda and something we're going to explore.

@ryantxu @toddtreece maybe you can expand a bit around your ideas with this "separate" api server and how wire fits or not fit in there. Dskit still on the agenda? I've heard things such as wire couldn't be used due to problem with database migrations and that you possibly want the api server to be "disconnected from the rest of the Grafana source code as much as possible". If we can solve the blockers with wire would it be feasible to use it within the api server?

Reply:
we paused the use of dskit modules/services for the apiserver because there was a short term requirement to use existing single tenant SQL for certain pieces. 

the last place we left it was that dskit modules can be used as long as each module either doesn't use wire, or if it does, it should have a separate wireset. the thinking there was that separate wiresets would help ensure that modules/services are decoupled.

Reply:
so for the moment, we can keep the slimmer version of the deps (thanks to @wbrowne) and we can iterate with a simpler interface / wire integration.